docs/standard/serialization/system-text-json/polymorphism.md

```diff
@@ -136,7 +139,607 @@ The following example shows the JSON that results from the preceding code:
+}
+
+public class FourDimensionalPoint : ThreeDimensionalPoint
+{
+    public int W { get; set; }
+}
+```
+
+```vb
+<JsonPolymorphic(
+    UnknownDerivedTypeHandling:=JsonUnknownDerivedTypeHandling.FallBackToBaseType)>
+<JsonDerivedType(GetType(ThreeDimensionalPoint))>
+Public Class BasePoint
+    Public Property X As Integer
+    Public Property Y As Integer
+End Class
+
+Public Class ThreeDimensionalPoint
+    Inherits BasePoint
+    Public Property Z As Integer
+End Class
+
+Public NotInheritable Class FourDimensionalPoint
+    Inherits ThreeDimensionalPoint
+    Public Property W As Integer
+End Class
+```
+
+The `FallBackToNearestAncestor` setting can be used to fall back to the contract of the nearest declared derived type:
+
+```csharp
+[JsonPolymorphic(
+    UnknownDerivedTypeHandling = JsonUnknownDerivedTypeHandling.FallBackToNearestAncestor)]
+[JsonDerivedType(typeof(BasePoint)]
+public interface IPoint { }
+
+public class BasePoint : IPoint { }
+
+public class ThreeDimensionalPoint : BasePoint { }
+```
+
+```vb
+<JsonPolymorphic(
+    UnknownDerivedTypeHandling = JsonUnknownDerivedTypeHandling.FallBackToNearestAncestor)>
+<JsonDerivedType(GetType(BasePoint)>
+Public Interface IPoint
+End Interface
+
+Public Class BasePoint
+    Inherits IPoint
+End Class
+
+Public Class ThreeDimensionalPoint
+    Inherits BasePoint
+End Class
+```
+
+With a configuration like the preceding example, the `ThreeDimensionalPoint` type will be serialized as `BasePoint`:
+
+```csharp
+// Serializes using the contract for BasePoint
+JsonSerializer.Serialize<IPoint>(new ThreeDimensionalPoint());
+```
+
+```vb
+' Serializes using the contract for BasePoint
+JsonSerializer.Serialize(Of IPoint)(New ThreeDimensionalPoint())
+```
+
+Falling back to the nearest ancestor admits the possibility of diamond ambiguity. Consider the following type hierarchy as an example:
+
+```csharp
+[JsonPolymorphic(
+    UnknownDerivedTypeHandling = JsonUnknownDerivedTypeHandling.FallBackToNearestAncestor)]
+[JsonDerivedType(typeof(BasePoint))]
+public interface IPoint { }
+
+public interface IPointWithTimeSeries : IPoint { }
+
+public class BasePoint : IPoint { }
+
+public class BasePointWithTimeSeries : BasePoint, IPointWithTimeSeries { }
+```
+
+```vb
+<JsonPolymorphic(
+    UnknownDerivedTypeHandling:=JsonUnknownDerivedTypeHandling.FallBackToNearestAncestor)>
+<JsonDerivedType(GetType(BasePoint))>
+Public Interface IPoint
+End Interface
+
+Public Interface IPointWithTimeSeries
+    Inherits IPoint
+End Interface
+
+Public Class BasePoint
+    Implements IPoint
+End Class
+
+Public Class BasePointWithTimeSeries
+    Inherits BasePoint
+    Implements IPointWithTimeSeries
+End Class
+```
+
+In this case, the `BasePointWithTimeSeries` type could be serialized as either `BasePoint` or `BasePointWithTimeSeries`. This ambiguity will cause the <xref:System.NotSupportedException> to be thrown when attempting to serialize an instance of `BasePointWithTimeSeries` as `IPoint`.
```

Comment:
```suggestion
In this case, the `BasePointWithTimeSeries` type could be serialized as either `BasePoint` or `IPointWithTimeSeries` since they are both direct ancestors. This ambiguity will cause the <xref:System.NotSupportedException> to be thrown when attempting to serialize an instance of `BasePointWithTimeSeries` as `IPoint`.
```